eq-author/src/components/CommentsPanel/index.test.js

```diff
@@ -194,4 +287,103 @@ describe("Comments Pane", () => {
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(deleteCommentButton).toHaveStyleRule("display: none;");
+  });
+
+  it("should render enabled delete button if comments exist && user === me", async () => {
+    const { getByTestId } = renderWithContext(<CommentsPanel />, { ...props });
+
+    await flushPromises();
+    const deleteCommentButton = getByTestId("btn-delete-comment-1");
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(deleteCommentButton).toHaveStyleRule("display: block");
+  });
+
+  it("should be able to delete an existing comment", async () => {
+    const { getByText, getByTestId } = renderWithContext(<CommentsPanel />, {
+      ...props,
+    });
+
+    await flushPromises();
+
+    const comment2 = getByText("Query comment2 body");
+    expect(comment2).toBeTruthy();
+
+    fireEvent.click(getByTestId("btn-delete-comment-1"));
+    await flushPromises();
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(comment2).toBeTruthy();
+  });
+
+  it("should hide Edit button if comments exist && comment user !== me", async () => {
+    const { getByTestId } = renderWithContext(<CommentsPanel />, { ...props });
+
+    await flushPromises();
+    const editCommentButton = getByTestId("btn-edit-comment-0");
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(editCommentButton).toHaveStyleRule("display: none;");
+  });
+
+  it("should render enabled Edit button if comments exist && user === me", async () => {
+    const { getByTestId } = renderWithContext(<CommentsPanel />, { ...props });
+
+    await flushPromises();
+    const editCommentButton = getByTestId("btn-edit-comment-1");
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(editCommentButton).toHaveStyleRule("display: block");
+  });
+
+  it("should show a save button for an edit comment area when edit button is clicked", async () => {
+    const { getByTestId } = renderWithContext(<CommentsPanel />, {
+      ...props,
+    });
+
+    await flushPromises();
+    fireEvent.click(getByTestId("btn-edit-comment-1"));
+
+    const editSaveBtn = getByTestId("btn-save-editedComment-1");
+
+    expect(queryWasCalled).toBeTruthy();
+    expect(editSaveBtn).toBeTruthy();
+    expect(editSaveBtn).toHaveStyle("display: inline-flex");
+  });
+
+  it("should update a comment", async () => {
+    const { getByText, getByTestId } = renderWithContext(<CommentsPanel />, {
+      ...props,
+    });
+
+    await flushPromises();
+    const editBtn = getByTestId("btn-edit-comment-1");
+    fireEvent.click(editBtn);
+
+    const editCommentTxtArea = getByTestId("edit-comment-txtArea-1");
+
+    fireEvent.change(editCommentTxtArea, {
+      target: { name: "name-1", value: "This is an edited comment" },
+    });
+    const editSaveBtn = getByTestId("btn-save-editedComment-1");
+    expect(editSaveBtn).toHaveStyle("display: inline-flex");
+
+    fireEvent.click(editSaveBtn);
+
+    await flushPromises();
+
+    expect(queryWasCalled).toBeTruthy();
```

Comment:
I am not sure what this is testing `expect(queryWasCalled).toBeTruthy()`. As any of the mocked requests will set `queryWasCalled` to `true`. Maybe change the query variables so one could be `deleteMutationWasCalled` and `updateMutationWasCalled` etc... This way we know that the right query/mutation was fired.

Reply:
makes sense - I've removed unnessary instances and renamed delete and update queries